backend/metadata_writer/src/metadata_writer.py

```diff
@@ -393,6 +397,9 @@ def is_kfp_v2_pod(pod) -> bool:
 
                 pods_with_written_metadata[obj.metadata.name] = None
 
-        except Exception as e:
-            import traceback
-            print(traceback.format_exc())
+        # If the for loop ended, a server-side timeout occurred. Continue watching.
+        pass
```

Comment:
this statement doesn't seem like it's necessary

Reply:
Technically yes, and originally there was just a comment here, but @hbelmiro suggested to add `pass` statement because otherwise it was confusing to which line the comment belongs.

Reply:
See https://github.com/kubeflow/pipelines/pull/11361#discussion_r1831807285